File: 636. Exclusive Time of Functions.cpp
class Solution {
public:
    vector<int> exclusiveTime(int n, vector<string>& logs) {
        vector < int > ans( n, 0 );
        stack < int > st;
        int lst = 0;
        for( string str: logs) {
            stringstream ss( str );
            string s_idx, s_status, s_end;
            getline( ss, s_idx, ':');
            getline( ss, s_status, ':');
            getline( ss, s_end, ':' );
            int idx = stoi(s_idx);
            int endTime = stoi(s_end);
            if( s_status == "start" ) {
                if( !st.empty() ) {
                    ans[st.top()] += ( endTime - lst );
                }
                lst = endTime;
                st.push(idx);
            } else {
                ans[st.top()] += ( endTime - lst + 1 );
                lst = endTime + 1;
                st.pop();
            }
            
        }
        return ans;
    }
};

File: Java/636. Exclusive Time of Functions.java
class Solution {
    public int[] exclusiveTime(int n, List<String> logs) {
        Stack < Integer > st = new Stack<> ();
        int[] res = new int[n];
        String[] str = logs.get(0).split(":");
        int idx = 1;
        int prv = Integer.parseInt(str[2]);
        st.push(Integer.parseInt(str[0]));
        while( idx < logs.size() ) {
            str = logs.get(idx).split(":");
            if( str[1].equals("start")) {
                if( !st.empty()) {
                    res[st.peek()] += Integer.parseInt(str[2]) - prv;
                }
                st.push(Integer.parseInt(str[0]));
                prv = Integer.parseInt(str[2]);
            } else {
                res[st.peek()] += Integer.parseInt(str[2]) - prv + 1;
                st.pop();
                prv = Integer.parseInt(str[2]) + 1;
            }
            idx++;
        }
                
        return res;
    }
}
